File: js/08_mar/LL_08_mar/helpers/SList.cpp
#include "SList.h"
#include <iostream>
#include <string>

SList::SList()
{
	head = NULL;
}

void SList::addFront(int val)
{
	node * tempNode = new node;
	tempNode -> value = val;
	tempNode -> next = head;
	head = tempNode;
}

void SList::printList()
{
	node * currNode = head;
	std::string lst = "[ ";

	while (currNode)
	{
		lst += std::to_string(currNode -> value) + ", ";

		currNode = currNode -> next;
	}

	lst += "]";

	std::cout << lst << std::endl;
}

int SList::getLength()
{
	node * currNode = head;
	int count = 0;

	while (currNode)
	{
		count++;

		currNode = currNode -> next;
	}

	return count;
}

int SList::getMax()
{
	if (!head)
	{
		std::cout << "List is empty" << std::endl;
		return -1;
	}
	node * currNode = head -> next;
	int max = head -> value;

	while (currNode)
	{
		if (currNode -> value > max) max = currNode -> value;

		currNode = currNode -> next;
	}

	return max;
}

int SList::getMin()
{
	if (!head)
	{
		std::cout << "List is empty" << std::endl;
		return -1;
	}
	node * currNode = head -> next;
	int min = head -> value;

	while (currNode)
	{
		if (currNode -> value < min) min = currNode -> value;

		currNode = currNode -> next;
	}

	return min;
}

int SList::getAvg()
{
	if (!head)
	{
		std::cout << "List is empty" << std::endl;
		return -1;
	}
	node * currNode = head -> next;
	int avg = head -> value,
		length = 1;

	while (currNode)
	{
		avg += currNode -> value;
		length++;
		currNode = currNode -> next;
	}

	return avg / length;
}

SList::~SList()
{
	node * p = head;
	node * q = head;

	while (q)
	{
		p = q;
		q = p -> next;

		if (q) delete p;
	}
}

File: js/08_mar/LL_08_mar/helpers/SList.java
package helpers;

import static java.lang.System.out;

public class SList {
	Node head;

	public SList() {
		this.head = null;
	}

	public void addFront(int val) {
		Node tempNode = new Node(val);
		tempNode.next = this.head;
		this.head = tempNode;
	}

	public void printList() {
		Node currNode = this.head;
		String lst = "[ ";

		while (currNode != null) {
			lst += currNode.value + ", ";

			currNode = currNode.next;
		}

		lst += "]";

		out.println(lst);
	}

	public int getLength() {
		Node currNode = this.head;
		int counter = 0;

		while (currNode != null) {
			counter++;

			currNode = currNode.next;
		}

		return counter;
	}

	public int getMax() {
		if (this.head == null) {
			out.println("List is empty");
			return -1;
		}

		Node currNode = this.head.next;
		int max = this.head.value;

		while (currNode != null) {
			if (currNode.value > max) {
				max = currNode.value;
			}

			currNode = currNode.next;
		}

		return max;
	}

	public int getMin() {
		if (this.head == null) {
			out.println("List is empty");
			return -1;
		}

		Node currNode = this.head.next;
		int min = this.head.value;

		while (currNode != null) {
			if (currNode.value < min) {
				min = currNode.value;
			}

			currNode = currNode.next;
		}

		return min;
	}

	public int getAvg() {
		if (this.head == null) {
			out.println("List is empty");
			return -1;
		}

		Node currNode = this.head.next;
		int avg = this.head.value,
			length = 1;

		while (currNode != null) {
			avg += currNode.value;
			length++;

			currNode = currNode.next;
		}

		return avg / length;
	}
}
